parsers/nmparser/get_model_output.go

```diff
@@ -67,48 +68,137 @@ func GetModelOutput(filePath string, verbose bool, noExt bool, noGrd bool, noCov
+
+		// stderr
+		nameCount = len(results.ParametersData[i].StdErr.Theta)
+		for n := nameCount; n < thetaCount; n++ {
+			results.ParametersData[i].StdErr.Theta = append(results.ParametersData[i].StdErr.Theta, 0)
 		}
-		if len(results.ParametersData[i].Estimates.Omega) != len(results.ParametersData[i].StdErr.Omega) {
-			results.ParametersData[i].StdErr.Omega = make([]float64, len(results.ParametersData[i].Estimates.Omega))
+		nameCount = len(results.ParametersData[i].StdErr.Omega)
+		for n := nameCount; n < omegaCount; n++ {
+			results.ParametersData[i].StdErr.Omega = append(results.ParametersData[i].StdErr.Omega, 0)
 		}
-		if len(results.ParametersData[i].Estimates.Sigma) != len(results.ParametersData[i].StdErr.Sigma) {
-			results.ParametersData[i].StdErr.Sigma = make([]float64, len(results.ParametersData[i].Estimates.Sigma))
+		nameCount = len(results.ParametersData[i].StdErr.Sigma)
+		for n := nameCount; n < sigmaCount; n++ {
+			results.ParametersData[i].StdErr.Sigma = append(results.ParametersData[i].StdErr.Sigma, 0)
+		}
+
+		// RandomEffectSD
+		nameCount = len(results.ParametersData[i].RandomEffectSD.Omega)
+		for n := nameCount; n < omegaCount; n++ {
+			results.ParametersData[i].RandomEffectSD.Omega = append(results.ParametersData[i].RandomEffectSD.Omega, 0)
 		}
+		nameCount = len(results.ParametersData[i].RandomEffectSD.Sigma)
+		for n := nameCount; n < sigmaCount; n++ {
+			results.ParametersData[i].RandomEffectSD.Sigma = append(results.ParametersData[i].RandomEffectSD.Sigma, 0)
+		}
+
+		// RandomEffectSDSE
+		nameCount = len(results.ParametersData[i].RandomEffectSDSE.Omega)
+		for n := nameCount; n < omegaCount; n++ {
+			results.ParametersData[i].RandomEffectSDSE.Omega = append(results.ParametersData[i].RandomEffectSDSE.Omega, 0)
+		}
+		nameCount = len(results.ParametersData[i].RandomEffectSDSE.Sigma)
+		for n := nameCount; n < sigmaCount; n++ {
+			results.ParametersData[i].RandomEffectSDSE.Sigma = append(results.ParametersData[i].RandomEffectSDSE.Sigma, 0)
+		}
+
+		// parameter names
+		nameCount = len(results.ParameterNames.Theta)
+		for n := nameCount; n < thetaCount; n++ {
+			results.ParameterNames.Theta = append(results.ParameterNames.Theta, fmt.Sprintf("Theta%v", n+1))
+		}
+		nameCount = len(results.ParameterNames.Omega)
+		for n := nameCount; n < omegaCount; n++ {
+			results.ParameterNames.Omega = append(results.ParameterNames.Omega, fmt.Sprintf("Omega%v", n+1))
+		}
+		nameCount = len(results.ParameterNames.Sigma)
+		for n := nameCount; n < sigmaCount; n++ {
+			results.ParameterNames.Sigma = append(results.ParameterNames.Sigma, fmt.Sprintf("Sigma%v", n+1))
+		}
+
 	}
 
-	for n := 0; n < len(results.ShrinkageDetails); n++ {
-		if len(results.ShrinkageDetails[n].Eta.SD) == 0 {
-			dim := 0
-			for i := range results.ParameterStructures.Omega {
-				if results.ParameterStructures.Omega[i] > 0 {
-					dim++
-				}
+	// shrinkage eta's
+	if etaCount > 0 {
```

Comment:
it looks like this should use a constructor function rather than all these sub loops. TOo much duplication

Reply:
a constructor for the float, like NewFloat? 
The dimension of the arrays are generally not known when the object is created. 
This code needs to be updated to change the 0's to a default value's, as part of that effort, I can investigate if the dimensions could be captured before instantiation